src/utils/connector.js

```diff
@@ -0,0 +1,140 @@
+import settings from '@/utils/settings'
+import { fetchData } from '@/utils/helpers/fetchData'
+
+function getContractsQueryBody (contractAgreementIdFilter) {
+  const body = {
+    '@context': { '@vocab': 'https://w3id.org/edc/v0.0.1/ns/' },
+    '@type': 'QuerySpec',
+    filterExpression: [
+      {
+        operandLeft: 'id',
+        operator: '=',
+        operandRight: contractAgreementIdFilter
+      }
+    ]
+  }
+  return body
+}
+
+function getNegotiationsQueryBody (currentPage, showConsumed, batchSize = settings.contractsPageSize) {
+  const offset = (currentPage - 1) * batchSize
+  let providerConditon = 'CONSUMER'
```

Comment:
Nitpick: sorry if I get a little picky with names, but we could also make this variable's name clearer. For instance:
```suggestion
  let offeringType = 'CONSUMER'
```